docs/source/_templates/sdk_submit_tutorial.rst

```diff
@@ -153,6 +167,51 @@ SHA-512 hash of the payload bytes.
     // Check if err is nil before continuing
     transactionHeaderBytes, err := proto.Marshal(&rawTransactionHeader)
 
+{% elif language == 'Rust' %}
+
+.. code-block:: rust
+
+    extern crate protobuf;
+    extern crate openssl;
+
+    use protobuf::Message
+    use protobuf::RepeatedField;
+
+    use openssl::sha::sha512;
+    use sawtooth_sdk::messages::transaction::TransactionHeader;
+
+    let mut txn_header = TransactionHeader::new();
+    txn_header.set_family_name(String::from("intkey"));
+    txn_header.set_family_version(String::from("1.0"));
+
+    let input_vec : Vec<String>
+      = vec![String::from(
+        "1cf1266e282c41be5e4254d8820772c5518a2c5a8c0c7f7eda19594a7eb539453e1ed7")];
+    let output_vec : Vec<String>
+      = vec![String::from(
+        "1cf1266e282c41be5e4254d8820772c5518a2c5a8c0c7f7eda19594a7eb539453e1ed7")];
+
+    txn_header.set_inputs(RepeatedField::from_vec(input_vec));
+    txn_header.set_outputs(RepeatedField::from_vec(output_vec));
+    txn_header.set_signer_public_key(signer.get_public_key()
+            .expect("Error retrieving Public Key").as_hex());
+    txn_header.set_batcher_public_key(signer.get_public_key()
+            .expect("Error retrieving Public Key").as_hex());
+    txn_header.set_payload_sha512(to_hex_string(&sha512(&payload_bytes).to_vec()));
+
```

Comment:
One missing parameter in all other languages is "nonce". There were few people on rocketchat who had issues referring to example available here, because they missed setting "nonce". Could you please consider that parameter?

Reply:
True, I didn't add it because it was not present in the other tutorials ( namely the python one, the one I followed more closely ) but I agree that it would beneficial to add it

Reply:
Since there is no guide in other examples, what should I use as nonce ?

Reply:
Omitting the nonce is a common source of bugs in writing Sawtooth clients.
Please use a random number (or sequence of random numbers) adding up to 128 bits. 64 bits is OK, but 128 is better.

Some client code uses dates, but that is not as reliable, especially in an environment with multiple hosts and multiple clients.